runtime/compiler/optimizer/J9TransformUtil.cpp

```diff
@@ -187,6 +187,58 @@ static bool isFieldOfJavaObject(TR::SymbolReference *symRef, TR::Compilation *co
    return false;
    }
 
+static bool isNullValueAtAddress(TR::Compilation *comp, TR::DataType loadType, uintptr_t fieldAddress, TR::Symbol *field)
+   {
+   TR_J9VMBase *fej9 = comp->fej9();
+
+   switch (loadType)
+      {
+      case TR::Int32:
```

Comment:
The types here were exhaustive for `avoidFoldingInstanceField()`, but for array element loads this function will also need to handle `Int8` and `Int16`

Reply:
You are absolutely right! But I am wondering why don't we check those types  in `J9::TransformUtil::transformIndirectLoadChainImpl() ` when we do the actual folding?

Reply:
Added two cases in `avoidFoldingInstanceField()`

Reply:
Interesting. I guess folding 8- and 16-bit array elements hasn't been particularly important for performance. I see no reason we shouldn't fold them, though that's not something you necessarily need to change in this PR. OTOH it would be really easy to add them. So I'll leave that up to you

But even with the final transformation refusing to fold `Int8` and `Int16`, the current logic would allow for an 8- or 16-bit load from a stable array to reach `isNullValueAtAddress()`, since it won't have been ignored yet. So these cases are necessary here either way. Thanks for adding them!

Reply:
Thanks! Good to know! I would like to limit the changes in this PR.